### ISSUE TITLE
Sync didn't work with Android Caldav

### ISSUE THREAD

**Issue by rehoehle** [owncloud/core#29979]
We have a  problem with a sync of an Android device with CalDav Sync.

### Expected behaviour
Calender should sync. Entries are available but doesn't appear on the device.

### Server configuration
**Operating system**:
Ubuntu 16.04

**Web server:**
Apache 2.4.18

**Database:**
MySQL 5.7.20

**PHP version:**
7.0.22

**ownCloud version:** (see ownCloud admin page)
10.0.4.-1.1 (From repo)

**Updated from an older ownCloud or fresh install:**
Update from OC 9

**Where did you install ownCloud from:**
Repository

I've raised the loglevel and debug to a higher level. But there is no error in the logs. Then i tried so fetch the data manually with curl and when i run the following command: 

curl -u xxx -X REPORT https://oc.ppcmusic.de/remote.php/dav/calendars/xxx/default/

i get the following error:
```"app":"webdav","method":"REPORT","url":"\/remote.php\/dav\/calendars\/xxxx\/personal\/","message":"Exception: 
{\"Message\":\"This should never happen (famous last words)\",\"Exception\":\"Sabre\\\\Xml\\\\ParseException\",\"Code\":0,\"Trace\":\"
#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/xml\\\/lib\\\/Element\\\/Base.php(86): Sabre\\\\Xml\\\\Reader->parseInnerTree()
#1 [internal function]: Sabre\\\\Xml\\\\Element\\\\Base::xmlDeserialize(Object(Sabre\\\\Xml\\\\Reader))
#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/xml\\\/lib\\\/Reader.php(247): call_user_func(Array, Object(Sabre\\\\Xml\\\\Reader))
#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/xml\\\/lib\\\/Reader.php(71): Sabre\\\\Xml\\\\Reader->parseCurrentElement()
#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/xml\\\/lib\\\/Service.php(118): Sabre\\\\Xml\\\\Reader->parse()
#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(715): Sabre\\\\Xml\\\\Service->parse('', '\\\/remote.php\\\/dav...', NULL)
#6 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpReport(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))
#7 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)
#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:REPORT', Array)
#9 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))
#10 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Server.php(256): Sabre\\\\DAV\\\\Server->exec()
#11 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()
#12 \\\/var\\\/www\\\/owncloud\\\/remote.php(165): require_once('\\\/var\\\/www\\\/ownclo...')
#13 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/xml\\\/lib\\\/Reader.php\",\"Line\":157,\"User\":\"xxxxxx\"}"}```


So i think the entries doesn't appear because there is an parse error. Does anyone have that problem in the past? We have checked the error with a new user and new entries but we can't find any problems.

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks a possibly related issue is #28826: Raised exceptions about REPORT operation when opening comments.

**Comment by rehoehle** [comment]
Perhaps but that here is a problem regarding the calender. And not the core or comments.

**Comment by RaSKoR** [comment]
Same bug for me, I fix it with this fix : https://central.owncloud.org/t/owncloud-10-0-4-caldav-sync-with-cellphone-workaround-needed/10797/3
It will be fix into OC 10.0.5
